File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/service/registry/RegisterFactory.java
package org.javamaster.invocationlab.admin.service.registry;

import java.util.List;
import java.util.Set;

/**
 * @author yudong
 */
public interface RegisterFactory {

    void addCluster(String cluster);

    Register remove(String cluster);

    Register get(String cluster);

    default void refreshService(List<String> interfaceNames, String cluster){}

    Set<String> getClusterSet();
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/service/registry/RegisterFactory.kt
package org.javamaster.invocationlab.admin.service.registry

/**
 * @author yudong
 */
interface RegisterFactory {
    fun addCluster(cluster: String)

    fun remove(cluster: String): Register

    fun get(cluster: String): Register

    fun refreshService(interfaceNames: List<String>, cluster: String) {}

    fun getClusterSet(): MutableSet<String>
}
